lib/googleauth/service_account.rb

```diff
@@ -211,16 +210,23 @@ def updater_proc
       protected
 
       # Creates a jwt uri token.
-      def new_jwt_token jwt_aud_uri, options = {}
+      def new_jwt_token jwt_aud_uri = nil, options = {}
         now = Time.new
         skew = options[:skew] || 60
         assertion = {
           "iss" => @issuer,
           "sub" => @issuer,
-          "aud" => jwt_aud_uri,
           "exp" => (now + EXPIRY).to_i,
           "iat" => (now - skew).to_i
         }
+
+        if @scope && jwt_aud_uri
+          raise ArgumentError, "Cannot specify both scope and aud"
```

Comment:
Question. Currently, `jwt_aud_uri` is basically _always_ set (for gRPC requests). That is, the Ruby grpc library always provides it as an FYI so that auth libraries can use it or not to resolve credentials. Thus, this code will currently always raise an error if custom scopes are provided.

I'm guessing the intent is that, if custom scopes are needed, we should simply include the scopes and omit the aud from the JWT's assertions. And thus, I _think_ this code should simply be:

```ruby
jwt_aud_uri = nil if @scope
```

Does that sound right?